File: packages/awesome_notifications_localizations/android/src/main/kotlin/me/carda/awesome_notifications_localizations/AwesomeNotificationsLocalizationsPlugin.kt
package me.carda.awesome_notifications_localizations

import io.flutter.embedding.engine.plugins.FlutterPlugin
import me.carda.android_awn_core.AwesomeMethodHandlerRegistry
import me.carda.android_awn_core.NotificationContentManager

/**
 * Localizations decorator plugin. It owns no method channel: on attach it
 * registers a content transformer (translates the notification at build time)
 * and a method handler (setLocalization/getLocalization) into the core's
 * decorator seams. The core stays unaware of localization.
 */
class AwesomeNotificationsLocalizationsPlugin : FlutterPlugin {

    private val manager = LocalizationManager.getInstance()
    private val transformer = LocalizationTransformer(manager)
    private val methodHandler = LocalizationMethodHandler(manager)

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        manager.init(binding.applicationContext)
        NotificationContentManager.register(transformer)
        AwesomeMethodHandlerRegistry.register(methodHandler)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        NotificationContentManager.unregister(transformer)
        AwesomeMethodHandlerRegistry.unregister(methodHandler)
    }
}

File: packages/awesome_notifications_localizations/ios/awesome_notifications_localizations/Sources/awesome_notifications_localizations/AwesomeNotificationsLocalizationsPlugin.swift
import Flutter
import UIKit
import IosAwnCore

/// Localizations decorator plugin (iOS).
///
/// Owns no method channel: on registration it plugs a content transformer
/// (translates the notification at build time) and a method handler
/// (setLocalization/getLocalization) into the core's decorator seams — the same
/// IosAwnCore singletons the base plugin's engine reads from (both plugins depend
/// on the same IosAwnCore package). The core stays unaware of localization.
public class AwesomeNotificationsLocalizationsPlugin: NSObject, FlutterPlugin {

    // Retained for the lifetime of the process so the registries keep working.
    private static let transformer = LocalizationTransformer()
    private static let methodHandler = LocalizationMethodHandler()

    public static func register(with registrar: FlutterPluginRegistrar) {
        NotificationContentManager.shared.register(transformer: transformer)
        AwesomeMethodHandlerRegistry.shared.register(handler: methodHandler)
    }
}
